control/runner.go

```diff
@@ -306,9 +306,87 @@ func (r *runner) HandleGomitEvent(e gomit.Event) {
 		if err != nil {
 			return
 		}
+	case *control_event.UnloadPluginEvent:
+		// On plugin unload,  find the key and pool info for the plugin being unloaded.
+		r.availablePlugins.Lock()
+		var pool strategy.Pool
+		var k string
+		for key, p := range r.availablePlugins.table {
+			tnv := strings.Split(key, ":")
+			if core.PluginType(v.Type).String() == tnv[0] && v.Name == tnv[1] && v.Version == p.Version() {
+				pool = p
+				k = key
+				break
+			}
+		}
+
+		r.availablePlugins.Unlock()
+		if pool == nil {
+			return
+		}
+		// Check for the highest lower version plugin and move subscriptions that
+		// are not bound to a plugin version to this pool.
+		plugin, err := r.pluginManager.get(fmt.Sprintf("%s:%s:%d", core.PluginType(v.Type).String(), v.Name, -1))
+		if err != nil {
+			return
+		}
+		newPool, err := r.availablePlugins.getOrCreatePool(plugin.Key())
+		if err != nil {
+			return
+		}
+		subs := pool.MoveSubscriptions(newPool)
+		// Start new plugins in newPool if needed
+		if newPool.Eligible() {
```

Comment:
What if newPool is not eligible. Do you still like to proceed all of other tasks such as deleting plugin...?